lib/modules/datasource/docker/index.ts

```diff
@@ -558,6 +594,68 @@ export class DockerDatasource extends Datasource {
     return null;
   }
 
+  @cache({
+    namespace: 'datasource-docker-architecture',
+    key: (
+      registryHost: string,
+      dockerRepository: string,
+      currentDigest: string
+    ) => `${registryHost}:${dockerRepository}@${currentDigest}`,
+    ttlMinutes: 1440 * 28,
+  })
+  public async getImageArchitecture(
+    registryHost: string,
+    dockerRepository: string,
+    currentDigest: string
+  ): Promise<string | null | undefined> {
+    try {
+      const manifestResponse = await this.getManifestResponse(
+        registryHost,
+        dockerRepository,
+        currentDigest,
+        'head'
+      );
+
+      if (
+        manifestResponse?.headers['content-type'] !== MediaType.manifestV2 &&
+        manifestResponse?.headers['content-type'] !== MediaType.ociManifestV1
+      ) {
+        return null;
+      }
+
+      const configDigest = await this.getConfigDigest(
+        registryHost,
+        dockerRepository,
+        currentDigest
+      );
+      if (!configDigest) {
+        return null;
+      }
+
+      const configResponse = await this.getImageConfig(
+        registryHost,
+        dockerRepository,
+        configDigest
+      );
+      if (configResponse) {
+        const architecture = configResponse.body.architecture ?? null;
+        logger.debug(
+          `Current digest ${currentDigest} relates to architecture ${
+            architecture ?? 'null'
+          }`
+        );
+
+        return architecture;
+      }
+    } catch (err) /* istanbul ignore next */ {
+      if (err.statusCode !== 404) {
```

Comment:
```suggestion
      if (err.statusCode !== 404 || err.message === PAGE_NOT_FOUND_ERROR) {
```

I think we need to rethrow `PAGE_NOT_FOUND_ERROR` error from `getAuthHeaders` to retry `node` -> `library/node`